app/src/main/java/com/lla/panicbutton/ui/fragments/SetupFragment.kt

```diff
@@ -21,5 +30,25 @@ class SetupFragment : Fragment(R.layout.fragment_setup) {
         nextButton.setOnClickListener {
             findNavController().navigate(R.id.action_setupFragment_to_panicButtonFragment)
         }
+
+        yourPhoneButton.setOnClickListener {
+        }
+    }
+
+    private suspend fun loadAudioFromInternalStorage() {
+        val intent: Intent = Intent()
+        intent.action = Intent.ACTION_GET_CONTENT
+        intent.type = "audio/mpeg"
+        startActivityForResult(intent, 1)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == 1 && resultCode == Activity.RESULT_OK) {
+            if (data != null && data.data != null) {
+                val audioFileUri: Uri? = data.data
+                // Now you can use that Uri to get the file path, or upload it, ...
+            }
+        }
```

Comment:
starting working on implementing uploads bit no clue what this code is honestly, still working on it, but didn't want to delete in case it's helpful to the person actually working on uploads.